### PR TITLE
feat: add filtering

### PR DESCRIPTION
**Pull request by samsja**

### CODE REVIEW

**Review by JoanFM — APPROVED** [review]

File: executor.py
Comment (alaeddine-13) [inline executor.py:120]: what will happen if filter is not in parameters ?
options are:
* return nothing
* return all
* raise an error

**Review by alaeddine-13 — COMMENTED** [review]
let's also wait until fix is merged in docarray and make sure to update docarray version for this indexer

File: executor.py
Comment (samsja) [reply executor.py:120]: For me it should return all like if the filter was None

**Comment by alaeddine-13** [comment]
let's pin the version of docarray

**Review by alaeddine-13 — APPROVED** [review]

**Pull request merged**
